core/src/main/java/com/digitalpebble/stormcrawler/proxy/ProxyManager.java

```diff
@@ -0,0 +1,30 @@
+/**
+ * Licensed to DigitalPebble Ltd under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * DigitalPebble licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.digitalpebble.stormcrawler.proxy;
+
+import org.apache.storm.Config;
+
+import java.io.FileNotFoundException;
+
+/**
+ * Proxy manager is an abstract class specification that details the required interface of a proxy manager
+ * */
+public interface ProxyManager {
+    void configure(Config conf) throws FileNotFoundException, IllegalArgumentException;
```

Comment:
see above - remove the explicit throws in the method signature and rethrow a runtime one in the code if needed
see for instance URLFilters